### PR TITLE
Amls 4997 - Bug fix for validation in renewal flow

### PR DESCRIPTION
**Pull request by slavvco**
Updated model of SendTheLargestAmountsOfMoney in renewal to display validation error message for whole group of controls rather than for first control only. 

## Related / Dependant PRs?

- PR in acceptance tests: [AMLS-4997](https://github.com/hmrc/amls-acceptance-test/pull/183)

## How Has This Been Tested?

- manually
- unit tests
- regression run

## Checklist

- [x] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/models/renewal/SendTheLargestAmountsOfMoney.scala [hunk 1]
```diff
@@ -17,47 +17,92 @@
...
-    }
-  }
+import play.api.libs.json.{Json, Reads, Writes}
+import utils.TraversableValidators
 
-}
+case class SendTheLargestAmountsOfMoney (countries: Seq[Country])
 
-object SendTheLargestAmountsOfMoney {
+private sealed trait SendTheLargestAmountsOfMoney0 {
```
Comment (drewmiley) [inline app/models/renewal/SendTheLargestAmountsOfMoney.scala]: Why this name?

File: app/models/renewal/SendTheLargestAmountsOfMoney.scala [hunk 2]
```diff
@@ -17,47 +17,92 @@
...
+   a: Path => RuleLike[A, Seq[String]],
+   cR: Rule[Seq[String], Seq[Country]]
+  ): Rule[A, SendTheLargestAmountsOfMoney] =
+    From[A] { __ =>
+
+      import TraversableValidators._
+      import utils.MappingUtils.Implicits.RichRule
 
-  implicit val format = Json.format[SendTheLargestAmountsOfMoney]
+      implicit val emptyToNone: String => Option[String] = {
```
Comment (drewmiley) [inline app/models/renewal/SendTheLargestAmountsOfMoney.scala]: Where is this used?

File: app/models/renewal/SendTheLargestAmountsOfMoney.scala [hunk 3]
```diff
@@ -17,47 +17,92 @@
...
+      case Seq(a, b) => Json.obj("country_1" -> a, "country_2" -> b)
+      case Seq(a) => Json.obj("country_1" -> a)
+      case _ => Json.obj()
     }
+  }
+}
+
+object SendTheLargestAmountsOfMoney {
+
+  private object Cache extends SendTheLargestAmountsOfMoney0
```
Comment (drewmiley) [inline app/models/renewal/SendTheLargestAmountsOfMoney.scala]: Is Cache the right name for this?

File: app/models/renewal/SendTheLargestAmountsOfMoney.scala (hunk 1 above)
Comment (slavvco) [reply app/models/renewal/SendTheLargestAmountsOfMoney.scala]: I just followed naming used in other places, but I could rename it to anything you want :)

File: app/models/renewal/SendTheLargestAmountsOfMoney.scala (hunk 2 above)
Comment (slavvco) [reply app/models/renewal/SendTheLargestAmountsOfMoney.scala]: this implicit is necessary for seqToOptionSeq[String] below

File: app/models/renewal/SendTheLargestAmountsOfMoney.scala (hunk 3 above)
Comment (slavvco) [reply app/models/renewal/SendTheLargestAmountsOfMoney.scala]: same as above just followed some other naming, but I could it change to something else if you want

**Pull request merged**